urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/json/JSONBoolean.java

```
@@ -0,0 +1,214 @@
+     * If the passed string does not represent a boolean the method throws
+     * a NumberFormatException.
+     *
+     * @param str The string to be parsed.
+     * @return The parsed boolean value.
+     * @throws NullPointerException If the passed string is null.
+     * @throws JSONException If the passed string does not represent a boolean.
+     **/
+    public static Boolean parseBoolean( String str, 
+					boolean caseSensitive )
+	throws NullPointerException,
+	       NumberFormatException {
+
+	if( str == null )
+	    throw new NullPointerException( "Cannot parse null to a boolean." );
+	
+	if( caseSensitive )
+	    str = str.toLowerCase();
+
+	
+	// This method only recognizes "true" and "false"!
+	if( str.equals("true") )
+	    return new Boolean( true );
+	else if( str.equals("false") )
+	    return new Boolean( false );
+	else
+	    throw new NumberFormatException( "The passed string does not represent a boolean value: '" + str + "'." );
+    }
+
+    /**
+     * This static method is the equivalent to Java's Boolean.parseBoolean(String). 
+     * It expects a String containing a number and converts it into a JSONBoolean 
+     * object.
+     *
+     * If the passed string does not represent a boolean the method throws
+     * a JSONException.
+     *
+     * @param str The string to be parsed.
+     * @return The parsed JSON number value.
+     * @throws NullPointerException If the passed string is null.
+     * @throws JSONException If the passed string does not represent a number.
+     **/
+    public static JSONBoolean parseJSONBoolean( String str, boolean caseSensitive )
+	throws NullPointerException,
+	       JSONException {
+
+	
+	try {
+	    // Try to convert string into a java value.
+	    Boolean bool = JSONBoolean.parseBoolean( str, caseSensitive );
+
+	    //return new JSONBoolean( bool );
+	    if( bool.booleanValue() )
+		return JSONValue.TRUE;
+	    else
+		return JSONValue.FALSE;
+	    
+	} catch( NumberFormatException e ) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)